packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -432,6 +443,8 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
                 connection.listen();
                 if (waitForConnection) {
                     waitForConnection.resolve(connection);
+                    this.isInteractiveTerminal = this.restored;
+                    this.onTermDidConnectProcessEmitter.fire(this);
```

Comment:
If this event is supposed to be fired when `interfacive` state changed when it should be called `onDidChangeInteractive`.

Or rename `interactive` to `connected`  and  rename an event `onDidChangeConnected`.

please never shorten, it our coding rule: https://github.com/eclipse-theia/theia/wiki/Coding-Guidelines#whole-words-names
But don't write over verbose names too, it is clear from `TerminalWidget` that events about terminal there is no need to add `Terminal` to each event name.